src/hydra-queue-runner/build-remote.cc

```diff
@@ -434,7 +434,11 @@ void State::buildRemote(ref<Store> destStore,
 
             auto now1 = std::chrono::steady_clock::now();
 
-            auto outputs = step->drv->outputPaths(*localStore);
+            StorePathSet outputs;
+            for (auto & i : step->drv->outputsAndOptPaths(*localStore)) {
```

Comment:
So I was going to suggest using `drv->outputsAndPaths()` here, but it seems to have disappeared mysteriously. (In a major git fail, `git log -p src/libstore/derivations.cc` shows it being added but not being removed...) @Ericson2314 Do you know why it was removed?

Reply:
It might have disappeared in a merge commit; I've noticed `git log -G` skipping that

Reply:
Does it mean that the patch is fine overall?

Reply:
I think this might have been intentionally removed because we didn't want people to unintentionally not support floating CA derivations.

The build-remote protocol needs to be changed for those anyways, so I wouldn't worry for now.